tests/PHPStan/Analyser/data/bug-5908.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Bug5908;
+
+use function PHPStan\Testing\assertType;
+
+function foo($mixed, string $s) {
+    ctype_digit($mixed);
+    assertType('mixed', $mixed);
+
+	if (ctype_digit($mixed)) {
+		assertType('float|int|numeric-string', $mixed);
+	}
+	assertType('mixed', $mixed);
+
+	if (ctype_digit($s)) {
+		assertType('numeric-string', $s);
+	}
+
+	if (ctype_digit($mixed)) {
+		assertType('float|int|numeric-string', $mixed);
+		return;
+	}
+	assertType('mixed~float|int', $mixed);
```

Comment:
It can still be `int` if `$mixed` is `42` for example. The difference between `ctype_digit` and `is_numeric` is described in the docs https://www.php.net/manual/en/function.ctype-digit.php, this will need custom extension that probably takes advantage of IntegerRangeType or something...